### PR TITLE
Update _Intrinsic for sphinx to capture the inner docstring

### PR DESCRIPTION
**Pull request by guilhermeleobas**
As title.

This also fixes https://github.com/numba/numba/issues/6308

### CODE REVIEW

File: numba/tests/test_extending.py [hunk 1]
```diff
@@ -1374,6 +1374,26 @@ def codegen(context, builder, signature, args):
...
+                pass  # do nothing, return None, should be turned into
+                # dummy value
+
+            return sig, codegen
+
+        self.assertEqual("numba.tests.test_extending", void_func.__module__)
+        self.assertEqual("void_func", void_func.__name__)
+        self.assertEqual("TestIntrinsic.test_docstring.<locals>.void_func",
+                         void_func.__qualname__)
+        self.assertDictEqual({'a': int}, void_func.__annotations__)
```
Comment (sklam) [inline numba/tests/test_extending.py:1394]: I wasn't aware these were added. I'll need to try in my next PR.

**Review by sklam — APPROVED** [review]
Thanks for the patch!

I have manually tested against the doctest case in #6308.

File: numba/tests/test_extending.py (hunk 1 above)
Comment (guilhermeleobas) [reply numba/tests/test_extending.py:1394]: `functools.update_wrapper` updates the following fields: `__module__`, `__name__`, `__qualname__`, `__annotations__` and `__doc__` ([ref](https://docs.python.org/3/library/functools.html#functools.update_wrapper))

Comment (sklam) [reply numba/tests/test_extending.py:1394]: oops, I mean `assertDictEqual`, not anything particular in your PR.  (I was rushing to finish up the review before starting a meeting and didn't realize the confusing comment.)

Comment (guilhermeleobas) [reply numba/tests/test_extending.py:1394]: Oh, ok 😄

**Pull request merged**